### ISSUE TITLE
flow check vs flow focus-check yields different results

### ISSUE THREAD

**Issue by avaly** [facebook/flow#5125]
Using flow `0.57.3`, I get the following output running `flow check`:

```
$ flow check
Found 0 errors
```

And when running `flow focus-check` for a file:

```
$ flow focus-check client/app/pages/builds/BuildsPage.react.js
[2017-10-17 22:16:59] Checking 1 files
[2017-10-17 22:16:59] executable=/home/avaly/.config/yarn/global/node_modules/flow-bin/flow-linux64-v0.57.3/flow
[2017-10-17 22:16:59] version=0.57.3
[2017-10-17 22:16:59] Parsing
[2017-10-17 22:16:59] Building package heap
[2017-10-17 22:16:59] Loading libraries
[2017-10-17 22:17:00] Resolving dependencies
[2017-10-17 22:17:00] to_merge: Focused: 1, Dependents: 1, Dependencies: 56
[2017-10-17 22:17:00] Calculating dependencies
[2017-10-17 22:17:00] Merging
[2017-10-17 22:17:00] Done
[2017-10-17 22:17:00] Checked set: Focused: 1, Dependents: 1, Dependencies: 56
Error: client/app/pages/builds/BuildsPage.react.js:39
 39: 			<AppPage sideBar={<ProjectsListConnector />} title={t('builds.title')}>
     			                  ^^^^^^^^^^^^^^^^^^^^^^^^^ React element `ProjectsListConnector`
 39: 			<AppPage sideBar={<ProjectsListConnector />} title={t('builds.title')}>
     			                  ^^^^^^^^^^^^^^^^^^^^^^^^^ props of React element `ProjectsListConnector`. This type is incompatible with
 25: 	InputProps: AppAuthenticatedPropsType,
     	            ^^^^^^^^^^^^^^^^^^^^^^^^^ object type. See: client/app/components/decorators/authenticated.js:25
  Property `account` is incompatible:
     25: 	InputProps: AppAuthenticatedPropsType,
         	            ^^^^^^^^^^^^^^^^^^^^^^^^^ property `account`. Property not found in. See: client/app/components/decorators/authenticated.js:25
     39: 			<AppPage sideBar={<ProjectsListConnector />} title={t('builds.title')}>
         			                  ^^^^^^^^^^^^^^^^^^^^^^^^^ props of React element `ProjectsListConnector`

...

Found 4 errors
```

I'm not sure exactly which result is correct, whether `flow check` missed an error or whether `flow focus-check` is  wrong.

**Comment by avaly** [comment]
Running on the same codebase with flow `0.56.0`: both `flow check` and `flow focus-check FILE` return the same errors from above.

**Comment by apsavin** [comment]
Maybe related to #5105

**Comment by apsavin** [comment]
@avaly see #5180

**Comment by avaly** [comment]
Indeed this issue seems to have been triggered by my usage of `resolve_dirname` option. It seems that changing the config like in https://github.com/facebook/flow/issues/5180#issuecomment-341893618 seems to fix the issue in flow `0.58.0`.
